File: Portfolio/Projects/employee-manager-android-application/CollegeAdmin/app/src/main/java/com/example/collegeadmin/MainActivity.java
package com.example.collegeadmin;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.RelativeLayout;

import com.example.collegeadmin.faculty.UpdateFaculty;
import com.example.collegeadmin.notice.AddNotice;
import com.example.collegeadmin.notice.DeleteNotice;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    CardView upLoadNotice,addGallery,addEbook,faculty,deleteNotice;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        upLoadNotice=findViewById(R.id.addNotice);
        upLoadNotice.setOnClickListener(this);
        addEbook=findViewById(R.id.addEbook);
        addEbook.setOnClickListener(this);
        addGallery=findViewById(R.id.addGallery);
        addGallery.setOnClickListener(this);
        faculty=findViewById(R.id.faculty);
        faculty.setOnClickListener(this);
        deleteNotice=findViewById(R.id.deleteNotice);
        deleteNotice.setOnClickListener(this);
    }
    @Override
    public void onClick(View v) {
        Intent intent;
        switch (v.getId())
        {

            case R.id.addNotice:
                 intent=new Intent(MainActivity.this, AddNotice.class);
                startActivity(intent);
                break;
            case R.id.addGallery:
                intent=new Intent(MainActivity.this,UploadImage.class);
                startActivity(intent);
                break;
            case R.id.addEbook:
                intent=new Intent(MainActivity.this,UploadPDF.class);
                startActivity(intent);
                break;
            case R.id.faculty:
                intent=new Intent(MainActivity.this, UpdateFaculty.class);
                startActivity(intent);
                break;
            case R.id.deleteNotice:
                intent=new Intent(MainActivity.this, DeleteNotice.class);
                startActivity(intent);
                break;

        }
    }
}

File: Portfolio/NotesApp/app/src/main/java/com/example/notesapp/MainActivity.kt
 package com.example.notesapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

 class MainActivity : AppCompatActivity(),NotesRVAdapter.INotesRVAdapter {
    lateinit var viewModel: NoteViewModel
    lateinit var recyclerView: RecyclerView
    lateinit var submit: Button
    lateinit var editnotes: EditText
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        recyclerView=findViewById(R.id.recyclerView)
        submit= findViewById(R.id.submit)
        editnotes=findViewById(R.id.editnotes)
        recyclerView.layoutManager= LinearLayoutManager(this)
        val adapter=NotesRVAdapter(this,this)
        recyclerView.adapter=adapter
        viewModel = ViewModelProvider(this,ViewModelProvider.AndroidViewModelFactory(application)).get(NoteViewModel::class.java)
        viewModel.allNotes.observe(this, Observer {list->
            list?.let {
                adapter.updateList(it)
            }
        })
        submit.setOnClickListener{
            val text = editnotes.text.toString()
            if(text.isNotEmpty())
            {
                viewModel.insertNote(Note(text))
                Toast.makeText(this,"Inserted",Toast.LENGTH_SHORT).show()
            }
        }
    }
     override fun onItemClicked(note: Note) {
viewModel.deleteNote(note)
Toast.makeText(this,"Deleted",Toast.LENGTH_SHORT).show()
     }
 }